ee/dataflatten/string_delimited.go

```diff
@@ -15,6 +15,75 @@ const (
 	DuplicateKeys
 )
 
+// StringDelimitedLineFunc returns a function that conforms to the interface expected by
+// dataflattentable.Table's execDataFunc property. Properties are grouped into a single
+// record per line of data by splitting the line into fields based on a separator.
+//
+// Headers are expected to either be provided, or set from the first line of data using the separator.
+// If no separator is provided, it's assumed that fields are separated by whitespace.
+// The first N lines of data can be skipped in case garbage is sent before the data.
+func StringDelimitedLineFunc(sep string, headers []string, skipN int) dataFunc {
```

Comment:
Should this move to the exec parsers?

Reply:
I first wrote this method for `flatpak`, but since I wasn't able to utilize it I still thought it might be of value to parse csv like data. Also since `snap` outputs data in that format I added that to use it as a model for `StringDelimitedLineFunc`.

I can understand wanting this in exec parsers, but I also feel like this will be a repeatable use case for binary output parsing.

Reply:
This has now been moved to the exec parsers.